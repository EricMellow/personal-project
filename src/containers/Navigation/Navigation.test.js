import {
  Navigation,
  AuthNavigation,
  UnauthNavigation,
  mapStateToProps,
  mapDispatchToProps
} from "./Navigation";
import { shallow } from "enzyme";
import React from "react";

describe('Navigation', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Navigation />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('AuthNavigation', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<AuthNavigation />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Navigation', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<UnauthNavigation />);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {

    it('should map authUser to props', () => {
      const mockState = {
        authUser: true,
        userId: 12345
      };
      const expected = {
        authUser: true
      };

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct params on removeAuthenticatedUser', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'REMOVE_USER'
      };
      mappedProps.removeUser();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch with the correct params on removeUserId', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'REMOVE_USER_ID'
      };

      mappedProps.removeId();

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
});