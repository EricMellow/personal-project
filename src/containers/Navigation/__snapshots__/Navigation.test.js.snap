// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation AuthNavigation should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    removeId={[Function]}
    removeUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/distance"
>
          Distance
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/tags"
>
          Tags
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/create"
>
          Create Activity
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button sign-out"
          onClick={[Function]}
          to="/"
>
          Sign Out
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/account"
>
          Account
</NavLink>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Distance",
          "className": "nav-button",
          "to": "/distance",
        },
        "ref": null,
        "rendered": "Distance",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Tags",
          "className": "nav-button",
          "to": "/tags",
        },
        "ref": null,
        "rendered": "Tags",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Create Activity",
          "className": "nav-button",
          "to": "/create",
        },
        "ref": null,
        "rendered": "Create Activity",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Sign Out",
          "className": "nav-button sign-out",
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Account",
          "className": "nav-button",
          "to": "/account",
        },
        "ref": null,
        "rendered": "Account",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/distance"
>
            Distance
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/tags"
>
            Tags
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/create"
>
            Create Activity
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button sign-out"
            onClick={[Function]}
            to="/"
>
            Sign Out
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/account"
>
            Account
</NavLink>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Distance",
            "className": "nav-button",
            "to": "/distance",
          },
          "ref": null,
          "rendered": "Distance",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Tags",
            "className": "nav-button",
            "to": "/tags",
          },
          "ref": null,
          "rendered": "Tags",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Create Activity",
            "className": "nav-button",
            "to": "/create",
          },
          "ref": null,
          "rendered": "Create Activity",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Sign Out",
            "className": "nav-button sign-out",
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Account",
            "className": "nav-button",
            "to": "/account",
          },
          "ref": null,
          "rendered": "Account",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navigation Navigation should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/distance"
>
          Distance
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/signin"
>
          Sign In
</NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav-button"
          to="/signup"
>
          Sign Up
</NavLink>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Distance",
          "className": "nav-button",
          "to": "/distance",
        },
        "ref": null,
        "rendered": "Distance",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Sign In",
          "className": "nav-button",
          "to": "/signin",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Sign Up",
          "className": "nav-button",
          "to": "/signup",
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/distance"
>
            Distance
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/signin"
>
            Sign In
</NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-button"
            to="/signup"
>
            Sign Up
</NavLink>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Distance",
            "className": "nav-button",
            "to": "/distance",
          },
          "ref": null,
          "rendered": "Distance",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Sign In",
            "className": "nav-button",
            "to": "/signin",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Sign Up",
            "className": "nav-button",
            "to": "/signup",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navigation should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
